crypto-pgp/src/main/kotlin/dev/msfjarvis/aps/data/crypto/GPGKeyManager.kt

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright © 2014-2021 The Android Password Store Authors. All Rights Reserved.
+ * SPDX-License-Identifier: GPL-3.0-only
+ */
+
+package dev.msfjarvis.aps.data.crypto
+
+import androidx.annotation.VisibleForTesting
+import com.github.michaelbull.result.Result
+import com.github.michaelbull.result.runCatching
+import com.proton.Gopenpgp.crypto.Crypto
+import java.io.File
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.withContext
+
+public class GPGKeyManager(filesDir: String, private val dispatcher: CoroutineDispatcher) :
+  KeyManager<GPGKeyPair> {
+
+  private val keyDir = File(filesDir, KEY_DIR_NAME)
+  override suspend fun addKey(key: GPGKeyPair): Result<String, Throwable> =
+    withContext(dispatcher) {
+      runCatching {
+        if (!keyDirExists()) error("Key directory does not exist")
+        val keyFile = File(keyDir, "${key.getKeyId()}.$KEY_EXTENSION")
+        if (keyFile.exists()) {
+          if (!keyFile.delete()) error("Couldn't delete existing key file with the same name")
+        }
```

Comment:
We should probably not replace existing keys like this. wdyt of `fun addKey(key: GPGKeyPair, replace: Boolean = false)` as the signature and this check changing into something like this:

```
if (keyFile.exists()) {
  if (!replace) {
    return Err(IllegalStateException("Pre-existing key was found for ${key.getKeyId()} but 'replace' is set to false")
  } else {
    if (!keyFile.delete()) error("Couldn't delete existing key file with the same name")
  }
}
```

Reply:
As an aside, I'm not a huge fan of invoking the `error` method from core business logic like this. We should be using a sealed class of errors like we did with [Git](https://github.com/android-password-store/Android-Password-Store/blob/9982562dc4e1ab4dbd058cf9d3c3c46fc598dec8/app/src/main/java/dev/msfjarvis/aps/util/git/ErrorMessages.kt) so that downstream consumers can special-case conditions like this one.